packages/react/src/transactions/useSelfFundedFallback.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  PendingSigningRequestError,
+  UserRejectedError,
+  WalletConnectionError,
+} from '@lens-protocol/domain/entities';
+import {
+  BroadcastingError,
+  SupportedTransactionRequest,
+} from '@lens-protocol/domain/use-cases/transactions';
+import { IEquatableError } from '@lens-protocol/shared-kernel';
+
+import { Operation, useOperation } from '../helpers/operations';
+import { SelfFundedProtocolCallRequest } from './adapters/SelfFundedProtocolCallRequest';
+import { usePayTransactionController } from './adapters/usePayTransactionController';
+
+/**
+ * An opaque data structure that encapsulates the data required to make a self-funded protocol call.
+ *
+ * @internal
+ */
+export type SelfFundedOperationRequest = SelfFundedProtocolCallRequest<SupportedTransactionRequest>;
+
+export interface ISelfFundedFallback {
+  /**
+   * The fallback request to be executed if the original request fails.
+   *
+   * See {@link useSelfFundedFallback} for more information.
+   */
+  fallback: SelfFundedOperationRequest;
+}
+
+/**
+ * Given a {@link BroadcastingError}, returns true if it implements the {@link ISelfFundedFallback} interface.
+ *
+ */
+export function supportsSelfFundedFallback(
+  error: IEquatableError,
+): error is BroadcastingError & ISelfFundedFallback {
+  return error instanceof BroadcastingError && error.fallback !== undefined;
+}
+
+export type SelfFundedOperation = Operation<
+  void,
+  PendingSigningRequestError | UserRejectedError | WalletConnectionError,
+  [SelfFundedOperationRequest]
+>;
+
+/**
+ * `useExploreProfiles` is an hook that let you retry a failed operation that could be self-funded.
```

Comment:
typo: `useSelfFundedFallback` is an hook that let you retry a failed operation that could be self-funded.